rusk/tests/services/multi_transfer.rs

```diff
@@ -175,6 +179,167 @@ fn wallet_transfer(
     );
 }
 
+/// Executes three different transactions in the same block, expecting only two
+/// to be included due to exceeding the block gas limit. The last of the
+/// transactions is a contract deployment, and is expected to deploy that
+/// contract, but then be reverted.
+fn wallet_transfer_deploy(
+    rusk: &Rusk,
+    wallet: &wallet::Wallet<TestStore, TestStateClient>,
+    amount: u64,
+) {
+    // Generate a receiver pk
+    let receiver = wallet
+        .phoenix_public_key(3)
+        .expect("Failed to get public key");
+
+    let mut rng = StdRng::seed_from_u64(0xdead);
+
+    let initial_balance_0 = wallet
+        .get_balance(0)
+        .expect("Failed to get the balance")
+        .value;
+    let initial_balance_1 = wallet
+        .get_balance(1)
+        .expect("Failed to get the balance")
+        .value;
+    let initial_balance_2 = wallet
+        .get_balance(2)
+        .expect("Failed to get the balance")
+        .value;
+
+    // Check the senders initial balance is correct
+    assert_eq!(
+        initial_balance_0, INITIAL_BALANCE,
+        "Wrong initial balance for the sender"
+    );
+    assert_eq!(
+        initial_balance_1, INITIAL_BALANCE,
+        "Wrong initial balance for the sender"
+    );
+    assert_eq!(
+        initial_balance_2, INITIAL_BALANCE,
+        "Wrong initial balance for the sender"
+    );
+
+    // Check the receiver initial balance is zero
+    assert_eq!(
+        wallet
+            .get_balance(3)
+            .expect("Failed to get the balance")
+            .value,
+        0,
+        "Wrong initial balance for the receiver"
+    );
+
+    let mut txs = Vec::with_capacity(3);
+
+    for i in 0..2 {
```

Comment:
Shouldn't you do 0..=2 if you want to include 3 txs?

Reply:
Right below there is a deployment transaction pushed, so there's three in total.

Reply:
right